reactor-netty-core/src/main/java/reactor/netty/ReactorNetty.java

```diff
@@ -211,23 +211,40 @@ public static String format(Channel channel, String msg) {
 		Objects.requireNonNull(msg, "msg");
 		if (LOG_CHANNEL_INFO) {
 			String channelStr;
+			StringBuilder result;
 			Connection connection = Connection.from(channel);
 			if (connection instanceof ChannelOperationsId) {
 				channelStr = ((ChannelOperationsId) connection).asLongText();
-				return new StringBuilder(1 + channelStr.length() + 2 + msg.length())
-						.append(CHANNEL_ID_PREFIX)
-						.append(channelStr)
-						.append("] ")
-						.append(msg)
+				if (channelStr.charAt(0) != TRACE_ID_PREFIX) {
+					result = new StringBuilder(1 + channelStr.length() + 2 + msg.length())
+							.append(CHANNEL_ID_PREFIX)
+							.append(channelStr)
+							.append(CHANNEL_ID_SUFFIX_1);
+				}
+				else {
+					result = new StringBuilder(channelStr.length() + 1 + msg.length())
+							.append(channelStr)
+							.append(CHANNEL_ID_SUFFIX_2);
+				}
+				return result.append(msg)
 						.toString();
 			}
 			else {
-				// Replace "[id: 0x" with '[' in order to keep it consistent with ChannelOperationsId#asLongText()
-				channelStr = channel.toString().substring(ORIGINAL_CHANNEL_ID_PREFIX_LENGTH);
-				return new StringBuilder(1 + channelStr.length() + 1 + msg.length())
-						.append(CHANNEL_ID_PREFIX)
-						.append(channelStr)
-						.append(' ')
+				channelStr = channel.toString();
+				if (channelStr.charAt(0) == CHANNEL_ID_PREFIX) {
```

Comment:
well it is always non empty and starts with `(` or `[`, but if you want we can add additional checks

Reply:
> if the id prefix is at the beginning of the string, ind == 0 and channelStr.substring(0, 0) is the empty String.

How this can happen you are in the branch where the string starts with `(`?